src/Router/Routing.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gacela\Router;
+
+/**
+ * @method static get(string $path, object|string $controller, string $action = '__invoke')
+ * @method static head(string $path, object|string $controller, string $action = '__invoke')
+ * @method static connect(string $path, object|string $controller, string $action = '__invoke')
+ * @method static post(string $path, object|string $controller, string $action = '__invoke')
+ * @method static delete(string $path, object|string $controller, string $action = '__invoke')
+ * @method static options(string $path, object|string $controller, string $action = '__invoke')
+ * @method static patch(string $path, object|string $controller, string $action = '__invoke')
+ * @method static put(string $path, object|string $controller, string $action = '__invoke')
+ * @method static trace(string $path, object|string $controller, string $action = '__invoke')
+ */
+final class Routing
+{
+    /**
+     * @param callable(RoutingConfigurator):void $fn
+     */
+    public static function configure(callable $fn): void
+    {
+        $routingConfigurator = new RoutingConfigurator();
+        $fn($routingConfigurator);
+
+        $route = self::findRoute($routingConfigurator);
+
+        if ($route) {
+            echo $route->run($routingConfigurator);
+        }
+    }
+
+    private static function findRoute(RoutingConfigurator $routingConfigurator): ?Route
+    {
+        foreach ($routingConfigurator->routes() as $route) {
+            if ($route->methodMatches()) {
+                $redirect = $routingConfigurator->redirects()[$route->path()] ?? null;
+                if ($redirect !== null) {
+                    return self::findRedirectRoute($redirect, $routingConfigurator);
+                }
+            }
+
+            if ($route->requestMatches()) {
+                return $route;
+            }
+        }
+
+        return null;
+    }
+
+    private static function findRedirectRoute(Redirect $redirect, RoutingConfigurator $routingConfigurator): ?Route
+    {
+        foreach ($routingConfigurator->routes() as $route) {
+            if ($route->isRedirected($redirect)) {
+                return $route;
```

Comment:
Instead of returning a route, I think you should do something more like 🤔 
```php
header('Location: ' . $route);
```

Reply:
My bad! You're right 100% I will change this. Thanks ! 🙏